bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/module/provider/AnnotationActionModuleTypeHelper.java

```diff
@@ -113,6 +114,21 @@ public Collection<ModuleInformation> parseAnnotations(String name, Object action
         return moduleInformation;
     }
 
+    public String getModuleIdFromMethod(String actionScope, Class<?> clazz, Method method) {
+        boolean hasOverloads = Arrays.stream(clazz.getDeclaredMethods())
+                .filter(m -> m.isAnnotationPresent(RuleAction.class)).map(Method::getName)
+                .filter(name -> name.equals(method.getName())).count() > 1;
+        String uid = actionScope + "." + method.getName();
+        if (hasOverloads) {
+            logger.debug("@RuleAction method {}::{} has overloads. Appending signature hash to UID.",
+                    clazz.getSimpleName(), method.getName());
+            String signature = Arrays.stream(method.getParameterTypes()).map(Class::getName)
+                    .collect(Collectors.joining(","));
+            uid += "#" + signature.hashCode();
```

Comment:
`.hashCode()` could create collisions, maybe something like

```
MessageDigest md5 = MessageDigest.getInstance("MD5");
Arrays.stream(methos,getParameterTypes()).forEach(t -> md5.update(t.getName().getBytes()));
uid += String.format("#%032x", new BigInteger(1, md5.digest()));
```

BTW: Why don't we do that for each method? That would make the need to check if there are overloads unnecessary.

Reply:
Good question. It is not really required, but thinking about it, it will be beneficial if one adds an overload for a method in the future as the id of the existing method won't change.
I will look into the "real" hashing.